client/src/audit/AuditTrail.tsx

```diff
@@ -0,0 +1,272 @@
+        style={{
+          maxHeight: '70vh',
+          overflow: 'auto',
+          top: '200px',
+        }}
+      >
+        <pre style={{ whiteSpace: 'pre-wrap', wordBreak: 'break-all' }}>
+          {modalContent}
+        </pre>
+      </Modal>
+      {/*<h2>Audit Trail Test</h2>*/}
+      <form onSubmit={onSearch} style={{ marginBottom: 16 }}>
+        <div style={{ display: 'inline-flex', alignItems: 'center' }}>
+          <select
+            value={source}
+            onChange={(e) => setSource(e.target.value as 'portal' | 'tapis')}
+            style={{ marginRight: 8 }}
+          >
+            <option value="portal">Most Recent User Session Data</option>
+            <option value="tapis">File Search Data</option>
+          </select>
+          <AutoComplete
+            value={username}
+            style={{ marginRight: 8, width: '200px' }}
+            options={
+              source === 'portal'
+                ? filteredUsernames.map((name) => ({
+                    value: name,
+                    label: name,
+                  }))
+                : []
+            }
+            onSelect={(value) => {
+              setUsername(value);
+            }}
+            onSearch={(searchText) => {
+              setUsername(searchText);
+            }}
+            placeholder="Username/File Name:"
+          />
+          <button
+            type="submit"
+            disabled={auditLoading || !username}
+            style={{ marginLeft: '8px' }}
+          >
+            {auditLoading ? 'Loading…' : 'Submit'}
+          </button>
+        </div>
+      </form>
+
+      {auditError && (
+        <div style={{ color: 'red' }}>Error: {auditError.message}</div>
+      )}
+
+      {auditData?.data && auditData.data.length === 0 && (
+        <div>No records found.</div>
+      )}
+
+      {auditData?.data && auditData.data.length > 0 && (
+        <table
+          style={{
+            width: '100%',
+            borderCollapse: 'collapse',
+            tableLayout: 'fixed',
+          }}
+        >
+          <thead>
+            <tr>
+              <th className={styles.headerCell} style={{ width: '50px' }}>
+                User
+              </th>
+              <th className={styles.headerCell} style={{ width: '50px' }}>
+                Date
+              </th>
+              <th className={styles.headerCell} style={{ width: '50px' }}>
+                Time
+              </th>
+              <th className={styles.headerCell} style={{ width: '100px' }}>
+                Portal
+              </th>
+              <th className={styles.headerCell} style={{ width: '200px' }}>
+                Action
+              </th>
+              <th className={styles.headerCell} style={{ width: '200px' }}>
+                Tracking ID
+              </th>
+              <th className={styles.headerCell} style={{ width: '100px' }}>
+                Details
+              </th>
+            </tr>
+          </thead>
```

Comment:
There's a lot of repetition here. You can probably simplify it like this:

```
<thead>
      <tr>
        {[
          { label: 'User', width: '50px' },
          { label: 'Date', width: '50px' },
          { label: 'Time', width: '50px' },
          { label: 'Portal', width: '100px' },
          { label: 'Action', width: '200px' },
          { label: 'Tracking ID', width: '200px' },
          { label: 'Details', width: '100px' },
        ].map((col) => (
          <th
            key={col.label}
            className={styles.headerCell}
            style={{ width: col.width }}
          >
            {col.label}
          </th>
        ))}
      </tr>
    </thead>
```

Reply:
And, if you think we might need to reuse this table styling in the future, do you think it would be helpful to create a Table component for the audit module? Like Jake's `FileListingTable.tsx`?